docs/getting_started/flyte_project_components.md

```diff
@@ -0,0 +1,101 @@
+---
+jupytext:
+  formats: md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+---
+
+# Flyte project components
+
+A Flyte project is a directory containing task and workflow code, internal Python source code, configuration files, and other artifacts needed to package up your code so that it can be run on a Flyte cluster.
+
+## Directory structure
+
+If you examine the project you created with `pyflyte init` in {doc}`"Creating a Flyte project <creating_a_flyte_project`, you'll see the following directory structure:
+
+```{code-block} bash
+my_project
+├── Dockerfile        # Docker image
+├── LICENSE
+├── README.md
+├── docker_build.sh   # Docker build helper script
+├── requirements.txt  # Python dependencies
+└── workflows
+    ├── __init__.py
+    └── example.py    # Example Flyte workflows
+```
+
+## Configuration files
+
+Flyte projects contain a `Dockerfile` and `requirements.txt` file that you can modify to suit the needs of your project.
+
+(getting_started_python_dependencies)=
+
+### `requirements.txt` Python dependencies
+
+You can specify pip-installable Python dependencies in your project by adding them to the
+`requirements.txt` file.
+
+```{note}
+We recommend using [pip-compile](https://pip-tools.readthedocs.io/en/latest/) to
+manage your project's Python requirements.
+```
+
+````{dropdown} See requirements.txt
+
+```{rli} https://raw.githubusercontent.com/flyteorg/flytekit-python-template/main/simple-example/%7B%7Bcookiecutter.project_name%7D%7D/requirements.txt
+:caption: requirements.txt
+```
+
+````
+
+(getting_started_dockerfile)=
+
+### Dockerfile
```

Comment:
TODO: Decide if we want to mention ImageSpec here, and if so, if we also want to actively discourage people from customizing Dockerfiles.

Reply:
I think we should remove this section and ONLY mention ImageSpec... maybe with a link at the end `for advanced image configurations, read on <building with Dockerfile> instead`